templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml

```diff
@@ -188,30 +187,46 @@ contents:
     EOF
     }
 
-    # add member cluster
+    # add member to cluster
     etcd_member_add() {
       echo "Updating etcd membership.."
       if [ -d "$ETCD_DATA_DIR" ]; then
         echo "Removing etcd data_dir $ETCD_DATA_DIR.."
         rm -rf $ETCD_DATA_DIR
       fi
 
-      RESPONSE=$(env ETCDCTL_API=3 $ETCDCTL --cert $ASSET_DIR/backup/etcd-client.crt --key $ASSET_DIR/backup/etcd-client.key --cacert $ASSET_DIR/backup/etcd-ca-bundle.crt \
-        --endpoints ${RECOVERY_SERVER_IP}:2379 member add $ETCD_NAME --peer-urls=https://${ETCD_DNS_NAME}:2380)
-
-       if [ $? -eq 0 ]; then
-         echo "$RESPONSE"
-         APPEND_CONF=$(echo "$RESPONSE" | sed -e '1,2d')
-         echo -e "\n\n#[recover]\n$APPEND_CONF" >> $ETCD_CONFIG
-       else
-         echo "$RESPONSE"
-         exit 1
-       fi
+      RESPONSE=$($ETCDCTL_WITH_TLS --endpoints ${RECOVERY_SERVER_IP}:2379 member add $ETCD_NAME --peer-urls=https://${ETCD_DNS_NAME}:2380)
+      if [ $? -eq 0 ]; then
+        echo "$RESPONSE"
+        APPEND_CONF=$(echo "$RESPONSE" | sed -e '1,2d')
+        echo -e "\n\n#[recover]\n$APPEND_CONF" >> $ETCD_CONFIG
+      else
+        echo "$RESPONSE"
+        exit 1
+      fi
     }
 
-    start_etcd() {
-      echo "Starting etcd.."
-      mv ${MANIFEST_STOPPED_DIR}/etcd-member.yaml $MANIFEST_DIR
+    # remove member from cluster by name
+    etcd_member_remove() {
+      NAME="$1"
+
+      if [ -z "$NAME" ]; then
+        echo "etcd_member_remove requires 1 argument NAME"
+        exit 1
+      fi
+
+      ID=$($ETCDCTL_WITH_TLS member list | awk -F "," "/\s${NAME}\,/"'{print $1}')
```

Comment:
If you are searching through the entire line, then, perhaps we don't need -F flag. Otherwise, you can match $3 to a string of space+name (something like "\s${NAME}").

This works for me:
```
etcdctl member list |  awk -F"," -v NAME="$NAME" ' $3 == " "NAME { print $1 }'
```

Reply:
Thanks for the input I think the above is fine. `/\s${NAME}\,/` is pretty straightforward and readable. I will leave this for now.